### PR TITLE
Transposed conv2d and wgrad split k examples

### PR DESCRIPTION
**Pull request by masahi**
Adds two examples of non-trivial API usage:
* `30_wgrad_split_k`: An example of wgrad with parallel split k, fp16 input and fp32 output. There are many details to get right.
* `31_transposed_conv2d`: Use the 2x2 strided Dgrad kernel to compute 2x upscaling conv2d transpose op (aka deconvolution).

This is ready for review @hwu36 @manishucsd

### CODE REVIEW

**Comment by masahi** [comment]
cc @hwu36

**Review by hwu36 — COMMENTED** [review]
i went over the parallel split-k part.  looks good to me, overall.

**Review by hwu36 — COMMENTED** [review]
Thank you for making deconv for us! looks good overall.

File: examples/30_wgrad_split_k/30_wgrad_split_k.cu [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+  std::cout << "split-k-slices: " << split_k_slices << std::endl;
+  std::cout << "workspace size: " << workspace_size << std::endl;
+
+  // Allocate workspace memory
+  cutlass::device_memory::allocation<uint8_t> workspace(workspace_size);
+
+  result.status = implicit_gemm.can_implement(arguments);
+  CUTLASS_CHECK(result.status);
+
+  // After the workspace is allocated, we point the GEMM destination pointer to the workspace.
```
[the remark this answers is not in the document]
Comment (masahi) [reply examples/30_wgrad_split_k/30_wgrad_split_k.cu:503]: you mean you want to remove "the"?

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -101,6 +101,8 @@ foreach(EXAMPLE
   27_ampere_3xtf32_fast_accurate_tensorop_gemm
   28_ampere_3xtf32_fast_accurate_tensorop_fprop
   29_ampere_3xtf32_fast_accurate_tensorop_complex_gemm
+  30_wgrad_split_k
+  31_transposed_conv2d
```
[the remark this answers is not in the document]
Comment (masahi) [reply examples/CMakeLists.txt]: Do you mean 30-33 (not 31-33) are already taken?

I renamed wgrad and transpose conv2d examples to 34 and 35, resp.

File: examples/30_wgrad_split_k/30_wgrad_split_k.cu (hunk 1 above)
[the remark this answers is not in the document]
Comment (hwu36) [reply examples/30_wgrad_split_k/30_wgrad_split_k.cu]: I mean GEMM=> WGRAD

File: examples/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (hwu36) [reply examples/CMakeLists.txt]: sorry. i need to make myself clear.  wgrad is 30.  transposed_conv2d is 34.

[the remark this answers is not in the document]
Comment (masahi) [reply examples/CMakeLists.txt]: ok restored `30_wgrad_split_k` and renamed to `34_transposed_conv2d`

**Comment by masahi** [comment]
@hwu36 All comments addressed.

**Comment by hwu36** [comment]
We need below to gracefully exit if the compiler or gpu is too old.

```
 int main(int argc, char const **args) {
+  bool notSupported = false;
+
+  // Ampere Tensor Core operations exposed with mma.sync are first available in CUDA 11.0.
+  //
+  // CUTLASS must be compiled with CUDA 11 Toolkit to run Conv2dFprop examples.
+  if (!(__CUDACC_VER_MAJOR__ > 11 || (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ >= 0))) {
+    std::cerr << "Ampere Tensor Core operations must be compiled with CUDA 11.0 Toolkit or later." << std::endl;
+    notSupported = true;
+  }
+
+  cudaDeviceProp props;
+  CUDA_CHECK(cudaGetDeviceProperties(&props, 0));
+
+  if (!(props.major > 8 || (props.major == 8 && props.minor >= 0))) {
+    std::cerr << "Ampere Tensor Ops must be run on a machine with compute capability at least 80."
+              << std::endl;
+    notSupported = true;
+  }
+
+  if (notSupported) {
+    return 0;
+  }
+
   Options options;
```

**Review by hwu36 — APPROVED** [review]

**Pull request merged**